### DISCUSSION TITLE
Live Gestures on Android Client

### DISCUSSION THREAD

**Discussion by Blueforcer** [rustdesk/rustdesk#6143] [Feature Request]
### Bug Description

It seems that any gesture will applied after the gesture was finished on the client.
This means, while scrolling i dont see where i am, only if i release the mouse.
Also if i draw something, the drawing is applied at the same speed, but afterwards.
Looks like playing a recording of the gesture on the client.
Is there any way to have "live" gestures like **Teamviewer** or **Anydesk** does on Android?

### How to Reproduce

Just scroll

### Expected Behavior

1:1 scrolling

### Operating system(s) on local side and remote side

Windows 11 -> Android

### RustDesk Version(s) on local side and remote side

1.2.3 -> 1.2.3

### Screenshots

https://github.com/rustdesk/rustdesk/assets/31169771/62f1e2e8-ef36-434d-819e-813be909a7b2

**Comment by XuanTian-LTD** [comment]
Indeed, there is no "real-time" operation, many things can not be done, Android -> Android also has the same problem, I hope the author can reconstruct the function!

**Comment by Blueforcer** [comment]
I decompiled the anydesk plugin and Smartphone Link app and think i found the classes wich do the trick.
Maybe it helps.

[InputInjector.zip](https://github.com/rustdesk/rustdesk/files/13227080/InputInjector.zip)
[AnyDeskGesture.zip](https://github.com/rustdesk/rustdesk/files/13227089/AnyDeskGesture.zip)
